glotaran/analysis/residual_calculation.py

```diff
@@ -60,18 +70,16 @@ def apply_weight(matrix, weight):
     penalties = []
     for label in problem_bag:
         data = problem_bag[label].data
-        size = problem_bag[label].global_axis.size
+        global_axis = problem_bag[label].global_axis
         matrices = matrix_jobs[label]
         weight = problem_bag[label].weight
         reduced_clp_labels[label] = []
         reduced_clps[label] = []
         residuals[label] = []
-        for i in range(size):
+        for i, index in enumerate(global_axis):
```

Comment:
Using `enumerate` and NOT using index doesn't make sense. Also, it is a code smell.